src/fides/api/service/saas_request/override_implementations/snap_request_overrides.py

```diff
@@ -0,0 +1,90 @@
+from fides.api.schemas.saas.shared_schemas import HTTPMethod, SaaSRequestParams
+from fides.api.service.connectors.saas.authenticated_client import AuthenticatedClient
+from fides.api.service.saas_request.saas_request_override_factory import (
+    SaaSRequestType,
+    register,
+)
+
+
+def signed_payload(given_email: str) -> str:
+    """This function is to take in the value of our identity email, and then ensure it is all lower case, and then hash it with SHA256"""
+    sub_email = given_email.lower()
+    hash_value = hashlib.sha256(sub_email.encode())
+    sig = hash_value.hexdigest()
+
+    return sig
+
+
+@register("snap_user_delete", [SaaSRequestType.DELETE])
+def snap_user_delete(
+    client: AuthenticatedClient,
+    param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    secrets: Dict[str, Any],
+) -> int:
+    rows_deleted = 0
+    params = {"limit": 500}
+    get_organizations = client.send(
+        SaaSRequestParams(
+            method=HTTPMethod.GET,
+            path="/v1/me/organizations?with_ad_accounts=true",
+            params=params,
+        )
+    )
+    assert get_organizations.ok
+    org_out = get_organizations.json()
+    ad_account_ids: List[str] = []
+    organizations = org_out.get("organizations", [])
+    for org_info in organizations:
+        organization = org_info.get("organization", {})
+        ad_accounts = organization.get("ad_accounts", [])
+        for ad_account in ad_accounts:
+            ad_account_ids.append(ad_account["id"])
+    for ad_account in ad_account_ids:
+        get_segments = client.send(
+            SaaSRequestParams(
+                method=HTTPMethod.GET,
+                path=f"/v1/adaccounts/{ad_account}/segments",
+                params=params,
+            )
+        )
+        assert get_segments.ok
+        ad_out = get_segments.json()
+        segments = ad_out.get("segments", [])
+        for segment_info in segments:
+            segment = segment_info.get("segment", {})
+            segment_id = segment.get("id")
+            for row_param_values in param_values_per_row:
+                email = row_param_values["email"]
+                payload = json.dumps(
+                    {
+                        "users": [
+                            {
+                                "schema": ["EMAIL_SHA256"],
+                                "data": [[signed_payload(email)]],
+                            }
+                        ]
+                    }
+                )
+                response = client.send(
+                    SaaSRequestParams(
+                        method=HTTPMethod.DELETE,
+                        path=f"/v1/segments/{segment_id}/users",
+                        headers={
+                            "Content-Type": "application/json",
+                        },
+                        body=payload,
+                    )
+                )
+                time.sleep(3)
```

Comment:
Do we need these sleeps? What was the issue you were trying to circumvent with these?

Reply:
I had some concerns about the rate limits for the API, it's 20 requests for an App key, but for a token it's 10. For our testing we wouldn't hit it, but for real use, I do worry about how many we'd be firing off.

Reply:
Adrian suggested tackling this in the config rather than sleep =) much better!